packages/smooth_app/lib/helpers/user_management_helper.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:openfoodfacts/utils/OpenFoodAPIConfiguration.dart';
+import 'package:smooth_app/database/dao_secured_string.dart';
+
+class UserManagementHelper {
+  UserManagementHelper._();
+
+  static const String USER_ID = 'user_id';
+  static const String PASSWORD = 'pasword';
+
+  /// Checks credentials and conditionally saves them
+  static Future<bool?> login(User user) async {
+    final bool rightCredentials;
+    try {
+      rightCredentials = await OpenFoodAPIClient.login(user);
+    } catch (e) {
+      // Returning null to show a error on the page
+      return null;
+    }
+
+    if (rightCredentials) {
+      OpenFoodAPIConfiguration.globalUser = user;
+      _putUser(user);
+    }
+
+    return rightCredentials && await _checkCredentialsInStorage();
+  }
+
+  /// Checks if the saved credentials are still valid
+  /// and mounts credentials for use in queries
+  static Future<bool?> checkAndReMountCredentials() async {
+    final String? userId = await DaoSecuredString.get(USER_ID);
+    final String? password = await DaoSecuredString.get(PASSWORD);
+
+    if (userId == null || password == null) {
+      return false;
+    }
+
+    final User user = User(userId: userId, password: password);
+
+    final bool rightCredentials;
+    try {
+      rightCredentials = await OpenFoodAPIClient.login(user);
+    } catch (e) {
+      // Returning null to show a error
+      return null;
+    }
```

Comment:
Already talked about `try catch`: if you catch the error in the method and just return `null`, you propagate nothing about the error.
I would return `Future<bool>` and throw an exception if relevant: in the exception I can have a message like 'server down' or 'time out' or 'you don't have internet connection`.

Reply:
I now simply throw an exception, but this is not yet visible to the end user